noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -97,8 +97,11 @@ pub comptime fn flatten_to_fields(name: Quoted, typ: Type, omit: [Quoted]) -> ([
     let mut aux_vars = &[];
 
     if omit.all(|to_omit| to_omit != name) {
-        if typ.is_field() | typ.as_integer().is_some() | typ.is_bool() {
-            // For field, integer and bool we just cast to Field and add the value to fields
+        if typ.is_field() {
+            // For field we just add the value to fields
+            fields = fields.push_back(name);
+        } else if typ.is_field() | typ.as_integer().is_some() | typ.is_bool() {
+            // For integer and bool we just cast to Field and add the value to fields
```

Comment:
I did this change such that we don't have the `as Field` in the resulting code when the value is already actually a field. I originally did this because I though it was causing compilation issue. After all it was not and but it seemed fine to just keep the change here.